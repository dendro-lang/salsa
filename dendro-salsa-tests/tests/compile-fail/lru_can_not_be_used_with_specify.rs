#[salsa::jar(db = Db)]
struct Jar(MyInput, lru_can_not_be_used_with_specify);

trait Db: salsa::DbWithJar<Jar> {}

#[salsa::input(jar = Jar)]
struct MyInput {
    field: u32,
}

#[salsa::tracked(jar = Jar, lru = 3, specify)]
fn lru_can_not_be_used_with_specify(db: &dyn Db, input: MyInput) -> u32 {
    input.field(db)
}

fn main() {}
