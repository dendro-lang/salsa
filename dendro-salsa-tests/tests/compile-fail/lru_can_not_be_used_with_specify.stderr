error: `specify` and `lru` cannot be used together
  --> tests/compile-fail/lru_can_not_be_used_with_specify.rs:11:38
   |
11 | #[salsa::tracked(jar = Jar, lru = 3, specify)]
   |                                      ^^^^^^^

error[E0412]: cannot find type `lru_can_not_be_used_with_specify` in this scope
 --> tests/compile-fail/lru_can_not_be_used_with_specify.rs:2:21
  |
2 | struct Jar(MyInput, lru_can_not_be_used_with_specify);
  |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope
